backends/p4tools/modules/testgen/targets/bmv2/program_info.cpp

```diff
@@ -129,6 +131,7 @@ std::vector<Continuation::Command> BMv2_V1ModelProgramInfo::processDeclaration(
         auto* egressPortVar =
             new IR::Member(IR::getBitType(TestgenTarget::getPortNumWidth_bits()),
                            new IR::PathExpression("*standard_metadata"), "egress_port");
+        cmds.emplace_back(Continuation::Guard(getPortConstraint(getTargetOutputPortVar())));
```

Comment:
Do not use a guard here. Just a normal comparison for the ingress port variable. It needs to be smaller than 7.